src/engine/bank02.asm

```diff
@@ -1,17 +1,1775 @@
+	push de
+	call DrawCursorEmpty_OppPlayArea
+	pop de
+
+;update x and y cursor positions
+	ld a, d
+	ld [wCursorDuelXPosition], a
+	ld a, e
+	ld [wCursorDuelYPosition], a
+
+; reset cursor blink
+	xor a
+	ld [wDuelCursorBlinkCounter], a
+
+.skip
+	ldh a, [hKeysPressed]
+	and A_BUTTON | B_BUTTON
+	jr z, .sfx
+	and A_BUTTON
+	jr nz, .a_pressed
+
+; B pressed
+	ld a, $ff ; cancel
+	call PlaySFXConfirmOrCancel
+	scf
+	ret
+	
+.a_pressed
+	call DrawCursor_OppPlayArea
+	ld a, $01
+	call PlaySFXConfirmOrCancel
+	scf
+	ret
+
+.sfx
+	ld a, [wcfe3]
+	or a
+	jr z, .draw_cursor
+	call PlaySFX
+
+.draw_cursor
+	ld hl, wDuelCursorBlinkCounter
+	ld a, [hl]
+	inc [hl]
+	and %00001111
+	ret nz ; only update cursor if blink's lower nibble is 0
+
+	ld a, $0f ; cursor byte
+	bit 4, [hl] ; only draw cursor if blink counter's fourth bit is not set
+	jr z, DrawByteInCursor_OppPlayArea
+; fallthrough
+
+; transforms cursor position into coordinates
+; in order to draw byte on menu cursor
+DrawCursorEmpty_OppPlayArea: ; 8741 (2:4741)
+	ld a, $00 ; white tile
+; fallthrough
+
+; draws in the cursor position
+; input:
+; a = tile byte to draw
+DrawByteInCursor_OppPlayArea: ; 8743 (2:4743)
+	ld e, a
+	ld a, 10
+	ld l, a
+	ld a, [wCursorDuelXPosition]
+	ld h, a
+	call HtimesL
+; h = 10 * cursor x pos
+
+	ld a, l
+	add 1
+	ld b, a
+	ld a, [wCursorDuelYPosition]
+	sla a
+	add 14
+	ld c, a
+; c = 11 + 2 * cursor y pos + 14
 
-Func_8211: ; 8211 (2:4211)
-	INCROM $8211, $833c
+; draw tile loaded in e
+	ld a, e
+	call WriteByteToBGMap0
+	or a
+	ret
 
-Func_833c: ; 833c (2:433c)
-	INCROM $833c, $8764
+DrawCursor_OppPlayArea: ; 8760 (2:4760)
+	ld a, $0f ; load cursor byte
+	jr DrawByteInCursor_OppPlayArea
+; fallthrough
 
+; seems to be function to deal with the Peek menu 
+; to select a prize card to view
 Func_8764: ; 8764 (2:4764)
-	INCROM $8764, $8932
+	call Set_OBJ_8x8
+	call LoadCursorTile
+; reset ce5c and ce56
+	xor a
+	ld [$ce5c], a
+	ld [$ce56], a
+
+; draw play area screen for the turn player
+	ldh a, [hWhoseTurn]
+	ld h, a
+	ld l, a
+	call DrawYourOrOppPlayArea_LoadTurnHolders
+
+.swap
+	ld a, [$ce56]
+	or a
+	jr z, .draw_menu
+; if ce56 != 0, swap turn
+	call SwapTurn
+	xor a
+	ld [$ce56], a
+
+.draw_menu
+	xor a
+	ld hl, PlayAreaMenuParameters
+	call InitializeMenuParameters
+	call DrawWideTextBox
+	
+	ld hl, YourOrOppPlayAreaData
+	call PlaceTextItems
+
+.loop1
+	call DoFrame
+	call HandleMenuInput ; await input
+	jr nc, .loop1
+	cp $ff
+	jr z, .loop1
+
+	call EraseCursor
+	ldh a, [hCurMenuItem]
+	or a
+	jp nz, Func_8883 ; jump if not first option
+
+; hCurMenuItem = 0
+	ld a, [wTurnHolder1]
+	ld b, a
+	ldh a, [hWhoseTurn]
+	cp b
+	jr z, .text
+
+; switch the play area to draw
+	ld h, a
+	ld l, a
+	call DrawYourOrOppPlayArea_LoadTurnHolders
+	xor a
+	ld [$ce56], a
+
+.text
+	call DrawWideTextBox
+	lb de, $01, $0e
+	call InitTextPrinting
+	ldtx hl, WhichCardWouldYouLikeToSeeText
+	call ProcessTextFromID
+
+	xor a
+	ld [wPrizeCardCursorPosition], a
+	lb de, $48, $c2
```

Comment:
This is just loading the function at $88c2 (it might be convenient to define a Func_88c2 label above its INCROM below, just so that this instruction gets a generic non-hardcoded value for eventual replacement)